.github/workflows/sanity-test.yml

```diff
@@ -1,11 +1,19 @@
 name: Sanity Test
 
-on: pull_request
+on: pull_request_target
```

Comment:
I used `pull_request_target` event, because the workflow needs to access repository secrets required for the test. More info: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Also due to this the workflow wasn't triggered in this PR, but in order to validate that this workflow works I created a PR against the source branch of this PR (`psturc:KONFLUX-3683`) here: https://github.com/psturc/konflux-ci/pull/6

Reply:
@gbenhaim @yftacherzog since `pull_request_target` [should be used with caution](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/), I think it would be a good idea to make running the test conditional, to avoid running something fishy in the CI.

One of the ways to do it could be using labels, i.e. something like [this](https://github.com/psturc/konflux-ci/commit/e54d35e93a662710227d5da10251863b0ad03216). Or another option I like would be to run tests when [auto-merge is enabled](https://medium.com/@all_the_things_data_ops/what-is-auto-merge-enabled-in-github-actions-4365f302b400). Wdyt?

Reply:
I'm ok with either.

Reply:
Ok I went with the [auto-merge option](https://github.com/konflux-ci/konflux-ci/pull/222/commits/757a190b93fb9cc3bd2e690f25182c69944d61c0) and enabled it for the repo together with **showing "update branch" button** (to make user aware of newer changes in the source branch):

<img width="407" alt="Screenshot 2024-07-25 at 8 40 21" src="https://github.com/user-attachments/assets/6ad01a5c-be4a-4db8-bae3-7af3b5a6934a">

We could also require the branch to be **always up-to-date before merge**, but I'm not too sure about enabling that:

<img width="720" alt="Screenshot 2024-07-25 at 8 47 33" src="https://github.com/user-attachments/assets/8a54ff8e-d8cb-4713-9b7a-4fc2264918ab">

Reply:
> We could also require the branch to be always up-to-date before merge, but I'm not too sure about enabling that:

Let's try it, and if it becomes messy then we can remove it.

Reply:
ok done